memory/alloydb/chat_message_history.go

```diff
@@ -0,0 +1,224 @@
+package alloydb
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/jackc/pgx/v5"
+	"github.com/tmc/langchaingo/internal/alloydbutil"
+	"github.com/tmc/langchaingo/llms"
+	"github.com/tmc/langchaingo/schema"
+)
+
+type ChatMessageHistory struct {
+	engine     alloydbutil.PostgresEngine
+	sessionID  string
+	tableName  string
+	schemaName string
+	overwrite  bool
+}
+
+var _ schema.ChatMessageHistory = &ChatMessageHistory{}
+
+// NewChatMessageHistory creates a new NewChatMessageHistory with options.
+func NewChatMessageHistory(ctx context.Context, engine alloydbutil.PostgresEngine, tableName string, sessionID string, opts ...ChatMessageHistoryStoresOption) (ChatMessageHistory, error) {
+	var err error
+	// Ensure required fields are set
+	if engine.Pool == nil {
+		return ChatMessageHistory{}, errors.New("alloyDB engine must be provided")
+	}
+	if tableName == "" {
+		return ChatMessageHistory{}, errors.New("table name must be provided")
+	}
+	if sessionID == "" {
+		return ChatMessageHistory{}, errors.New("session ID must be provided")
+	}
+	cmh := ChatMessageHistory{
+		engine:    engine,
+		tableName: tableName,
+		sessionID: sessionID,
+	}
+	cmh, err = applyChatMessageHistoryOptions(cmh, opts...)
+	if err != nil {
+		return ChatMessageHistory{}, fmt.Errorf("applyChatMessageHistoryOptions(): %w", err)
+	}
+	err = cmh.validateTable(ctx)
+	if err != nil {
+		return ChatMessageHistory{}, fmt.Errorf("validateTable(): %w", err)
+	}
+	return cmh, nil
+}
+
+// validateTable validates if a table with a specific schema exist and it
+// contains the required columns.
+func (c *ChatMessageHistory) validateTable(ctx context.Context) error {
+	tableExistsQuery := fmt.Sprintf(`SELECT EXISTS (
+		SELECT FROM information_schema.tables 
+		WHERE table_schema = '%s' AND table_name = '%s');`,
+		c.schemaName, c.tableName)
+	var exists bool
+	err := c.engine.Pool.QueryRow(ctx, tableExistsQuery).Scan(&exists)
+	if err != nil {
+		return fmt.Errorf("error validating table %s: %w", c.tableName, err)
+	}
+	if !exists {
+		return fmt.Errorf("table '%s' does not exist in schema '%s'", c.tableName, c.schemaName)
+	}
+
+	requiredColumns := []string{"id", "session_id", "data", "type"}
+	for _, reqColumn := range requiredColumns {
+		columnExistsQuery := fmt.Sprintf(`SELECT EXISTS (
+			SELECT 1 FROM information_schema.columns 
+			WHERE table_schema = '%s' AND table_name = '%s' AND column_name = '%s'
+		);`, c.schemaName, c.tableName, reqColumn)
```

Comment:
int: Could this be optimized so we don't need 4 different queries? Can we just get all columns and validate the ones we want are present? Or get all 4 in a single query?

Reply:
This was part of another PR, we can have this fixes in a separate one, as this one its supposed to be the tests only!

Reply:
fix addressed in PR https://github.com/averikitsch/langchaingo/pull/19